### PR TITLE
LSE and Ply

### PR DESCRIPTION
**Pull request by the-headless-ghost**
~Currently, I'm hosting a branch with an updated version of Ply. As soon as I get push right, we could push it to a branch of the Ply repository.~

- `liqwid-plutarch-extra`
  - `ply-core` and `ply-plutarch` dependency
  - `PlyArg` instance for `AssetClass` and `ExtendedAssetClass`
  - `PlyArgOf` implementation for `PAssetClassData` and `PExtendedAssetClass`
- reintroduced `liqwid-script-export`

### CODE REVIEW

File: liqwid-script-export/example/Main.hs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Example usage of 'liqwid-script-export'.
+-}
+module Main (main) where
+
+import Data.Default (def)
+import Data.Map (fromList)
+import Data.Text (unpack)
+import Plutarch (compile)
+import Plutarch.Api.V2 (PValidator)
```
Comment (peter-mlabs) [inline liqwid-script-export/example/Main.hs]: Is this still using an old version of plutarch? Or did this module just not get updated?

File: liqwid-script-export/example/README.md
Comment (peter-mlabs) [inline liqwid-script-export/example/README.md:1]: Did we test that this still works?

File: liqwid-script-export/liqwid-script-export.cabal [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+--   import:         lang, deps
+--   hs-source-dirs: export-stat
+--   main-is:        Main.hs
+--   build-depends:  liqwid-script-export
+
+-- executable example
+--   import:         lang, deps
+--   hs-source-dirs: example
+--   main-is:        Main.hs
+--   build-depends:  liqwid-script-export
```
Comment (peter-mlabs) [inline liqwid-script-export/liqwid-script-export.cabal]: We should either:
- Remove these altogether (the cabal stanzas and the relevant files)
- Uncomment/test these

File: liqwid-script-export/src/ScriptExport/ScriptInfo.hs
Comment (peter-mlabs) [inline liqwid-script-export/src/ScriptExport/ScriptInfo.hs:1]: I see some @deprecated haddock comments in here. Should we be using the `{-# DEPRECATED ... #-}` [pragma](https://downloads.haskell.org/~ghc/6.4/docs/html/users_guide/pragmas.html) instead (or in addition to)?

**Review by peter-mlabs — COMMENTED** [review]
Besides the "deprecated" stuff, I think its okay if you make tickets for the remaining comments if you want

File: liqwid-script-export/liqwid-script-export.cabal (hunk 2 above)
Comment (the-headless-ghost) [reply liqwid-script-export/liqwid-script-export.cabal]: Uff... I forgot to uncomment these!

File: liqwid-script-export/example/README.md
Comment (the-headless-ghost) [reply liqwid-script-export/example/README.md:1]: Yes, it works now :-)

**Pull request merged**